src/dev/notice/generate_notice_text.js

```diff
@@ -0,0 +1,57 @@
+import vfs from 'vinyl-fs';
+
+import { REPO_ROOT } from '../constants';
+
+const NOTICE_COMMENT_RE = /\/\*[\s\n\*]*@notice([\w\W]+?)\*\//g;
+const NEWLINE_RE = /\r\n|[\n\r\u2028\u2029]/g;
+
+export async function generateNoticeText() {
+  const files = vfs.src([
+    '**/*.{js,less,css}',
+  ], {
+    cwd: REPO_ROOT,
+    nodir: true,
+    ignore: [
+      '{node_modules,build,target,dist}/**',
+      'packages/*/{node_modules,build,target,dist}/**',
+    ]
+  });
+
+  const noticeComments = [];
+  await new Promise((resolve, reject) => {
+    files
+      .on('data', (file) => {
+        const source = file.contents.toString('utf8');
+        let match;
+        while ((match = NOTICE_COMMENT_RE.exec(source)) !== null) {
+          noticeComments.push(match[1]);
+        }
+      })
+      .on('error', reject)
+      .on('end', resolve);
+  });
+
+  let noticeText = '';
+  noticeText += 'Kibana\n';
+  noticeText += `Copyright 2012-${(new Date()).getUTCFullYear()} Elasticsearch B.V.\n`;
+
+  for (const comment of noticeComments.sort()) {
```

Comment:
I just want the license comments to maintain their order regardless of where they are on the file system so that `--frozen` won't fail if a filename was just changed to cause a sorting difference. I'm also not 100% sure that `vinyl-fs` will always send file objects in the same order.

Reply:
Okay, got it, thanks

>  I'm also not 100% sure that vinyl-fs will always send file objects in the same order.

Yeah, me too